src/workspace/mod.rs

```diff
@@ -2226,6 +2318,145 @@ mod test {
     );
   }
 
+  #[test]
+  fn test_root_member_patch() {
+    let workspace_dir = workspace_for_root_and_member_with_fs(
+      json!({
+        "patch": ["../dir"],
+      }),
+      json!({
+        "patch": [
+          "../../dir"
+        ],
+      }),
```

Comment:
I find this confusing. From what I can tell, you can specify a path to a directory that contains one or more packages that should be used for "patching". But to see which packages will actually be patched I need to go to that directory and look up all the configs to figure out the names.

Is there a reason why you went with a list for patch directories instead of having to specify a mapping between package name and its patch directory instead?

Reply:
This seems confusing because the test has the name `dir`. Usually the directory name would be something like `../deno_std/path` or something like that. The user having to specify the package name in addition to the directory just seems like unnecessary bookkeeping since we can already tell the package name from the directory?

Reply:
Okayi guess that's reaaonable